Plutarch/DataRepr/Internal.hs

```diff
@@ -174,6 +175,21 @@ type family PLabelIndex (name :: Symbol) (as :: [PLabeledType]) :: Nat where
 type family PUnLabel (a :: PLabeledType) :: PType where
   PUnLabel (name ':= a) = a
 
+type PUnLabelAll :: [PLabeledType] -> [PType]
+type family PUnLabelAll l where
+  PUnLabelAll '[] = '[]
+  PUnLabelAll (x ': xs) = PUnLabel x ': PUnLabelAll xs
+
+instance
+  {-# OVERLAPPABLE #-}
+  ( All PIsData bs
+  , bs ~ PUnLabelAll xs
+  ) =>
+  PIsData (PDataRecord xs)
+  where
+  pfromData x = punsafeBuiltin PLC.UnListData # pforgetData x
+  pdata x = punsafeBuiltin PLC.ListData # x
```

Comment:
I don't think this is correct, `PDataRecord` is not a `List` but a `Constr`, and this would be inconsistent with the existing `PIsDataRepr` instances, where `pfromData` is a no-op.

Reply:
`PDataRecord` is actually a `List`. `PDataSum` is `Constr`.

Reply:
See `PlutusType` instance for `PDataRecord`